index/repo.go

```diff
@@ -0,0 +1,50 @@
+package index
+
+import (
+	"github.com/filecoin-project/dagstore/shard"
+)
+
+// Repo is the central index repository object that manages full indices and
+// manifests.
+type Repo struct {
+	FullIndexRepo
+	ManifestRepo
+}
+
+type Stat struct {
+	Exists bool
+	Size   uint64
+}
+
+type FullIndexRepo interface {
+	// GetFullIndex returns the full index for the specified shard.
+	GetFullIndex(key shard.Key) (FullIndex, error)
+
+	// AddFullIndex persists a full index for a shard.
+	AddFullIndex(key shard.Key, index FullIndex) error
```

Comment:
I think we will need to add a `Write` method to `FullIndex` in order for `AddFullIndex` to be able to write the index to disk

Reply:
Makes sense. We do want to transition to using the CARv2 Index abstraction (import from CARv2), which I believe provides that: https://github.com/ipld/go-car/blob/2a49cbd958b0c6443dd1e9b518814199bf50842d/v2/internal/index/index.go#L37-L43